cli/api/ioutil2/ioutil.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package ioutil2 provides extra functionality along similar lines to io/ioutil.
+package ioutil2
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+// Write file to temp and atomically move when everything else succeeds.
+func WriteFileAtomic(filename string, data []byte, perm os.FileMode) error {
+	dir, name := path.Split(filename)
+	f, err := ioutil.TempFile(dir, name)
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(data)
```

Comment:
Don't we want to exit immediately on error (and cleanup temp file)? This code will execute every subsequent operation after an error.

Reply:
The error handling is a little weird, but looks correct, and I don't want to rewrite this external file